scout/adapter/mongo/variant.py

```diff
@@ -552,10 +553,18 @@ def check_causatives(self, case_obj=None, institute_obj=None, limit_genes=None):
                 0:4
             ]  # example: [ "17", "7577559", "G" "A"]
 
+            positional_variant_ids = set()
             for variant_type in ["clinical", "reseach"]:
                 positional_variant_ids.add(generate_md5_key(other_var_simple + [variant_type]))
 
-        return self.match_affected_gt(case_obj, institute_obj, positional_variant_ids, limit_genes)
```

Comment:
The problem was that when invoked from causatives controller case_obj here is None. Then `match_affected_gt` queries for variants in all cases of the institute_obj, also those that are not causative.